cogs5e/models/homebrew/bestiary.py

```diff
@@ -362,33 +372,73 @@ def parse_critterdb_traits(data, key):
 
         if overrides:
             for override in overrides:
-                attacks.append({'name': override.group(1) or name,
-                                'attackBonus': override.group(2) or None, 'damage': override.group(3) or None,
-                                'details': desc})
+                if override.group('simple'):
+                    attacks.append(Attack.from_dict({
+                        'name': override.group(2) or name,
+                        'attackBonus': override.group(3) or None,
+                        'damage': override.group(4) or None,
+                        'details': desc
+                    }))
+                elif (freeform_override := override.group('freeform')):
+                    try:
+                        attack_yaml = yaml.safe_load(freeform_override)
+                    except yaml.YAMLError:
+                        raise ExternalImportError(
+                            f"Monster had an invalid automation YAML ({data['name']}: {name})")
+                    if not isinstance(attack_yaml, list):
+                        attack_yaml = [attack_yaml]
+                    for atk in attack_yaml:
+                        if isinstance(atk, dict):
+                            atk['name'] = atk_name = atk.get('name') or name
+                            try:
+                                attacks.append(Attack.from_dict(atk))
+                            except KeyError:
```

Comment:
This exception bound may not catch all possible exceptions

Reply:
Noted, have changed this to `except Exception:` in the latest commit. Thanks